CMakeLists.txt

```diff
@@ -118,25 +127,54 @@ add_definitions (-DSG_DATA_TYPE=double)
         elseif ("${COMPILER}" STREQUAL "xl")
             set(CMAKE_C_COMPILER xlc_r)
             set(CMAKE_CXX_COMPILER xlc++_r)
-            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -qenablevmx")
-            set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -qenablevmx")
 
         #intel
         elseif ("${COMPILER}" STREQUAL "intel")
+
+            #Check for intel platform variable set in user cmake call
+            if (NOT DEFINED INTEL_PLATFORM)
+                message("No intel platform defined.")
+            else()
+                #Set platform
+                ####
+                string( TOLOWER "${INTEL_PLATFORM}" INTEL_PLATFORM )
+
+                #skylake
+                if("${INTEL_PLATFORM}" STREQUAL "skylake")
+                    set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS} -march=skylake-avx512")
+
+                #cross-platform with AVX
+                elseif("${INTEL_PLATFORM}" STREQUAL "avx_crossplatform")
+                    set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS} -xCOMMON-AVX512")
+                
+                #Non-AVX platforms
+                elseif("${INTEL_PLATFORM}" STREQUAL "non_avx")
+                    #No specified flags
+                else()
+                    message("Platform was not recognized, no options passed.")
+                endif()
+            endif()
+
+            #Set compiler based on options
```

Comment:
This might be a separate option for MPICC as the compiler that supersedes the compiler?